Quantum-Control-Applications/Superconducting/Multiple Flux Tunable Transmons/Use Case 2 - Two Qubit Randomized Benchmarking/README.md

```diff
@@ -0,0 +1,238 @@
+
+## Prerequisites
+- Calibrated Single Qubit Gates
+- Calibrated CZ Gate
+- Calibrated Measurement Protocol for Qubit State Discrimination
+
+# Quick User Guide
+For a quick implementation just clone the [*qua-libs repository*](https://github.com/qua-platform/qua-libs/) or download the [*Use Case 2 - Two Qubit Randomized Benchmarking*](https://github.com/qua-platform/qua-libs/tree/2qb-RB-usecase/Quantum-Control-Applications/Superconducting/Multiple%20Flux%20Tunable%20Transmons/Use%20Case%202%20-%20Two%20Qubit%20Randomized%20Benchmarking) folder and edit the [*two_qubit_rb_example.py*](https://github.com/qua-platform/qua-libs/blob/2qb-RB-usecase/Quantum-Control-Applications/Superconducting/Multiple%20Flux%20Tunable%20Transmons/Use%20Case%202%20-%20Two%20Qubit%20Randomized%20Benchmarking/two_qubit_rb_example.py) python file.
+
+## Single Qubit Gates
+The function for the single qubit gates requires that the operation "x" was previously calibrated by the user and corresponds to a pi-pulse on the target qubits. The *amp=x* condition inside the *baker.play* statement allows to scale the amplitude of the pulse. Together with the first *baker.frame_rotation_2pi* it allows the *baker.play* statement to act as X and Y gates by shifting the frame of the control signal, thus realizing rotations around the x- and y-axis. The second *baker.frame_rotation_2pi* resets the frame and additionally allows for rotations around the z-axis.
+```python
+def bake_phased_xz(baker: Baking, q, x, z, a):
+    element = f"qubit{q}_xy"
+    baker.frame_rotation_2pi(-a, element)
+    baker.play("x", element, amp=x)
+    baker.frame_rotation_2pi(a + z, element)
+```
+
+## Two-Qubit Gate (CZ)
+The use-case is designed for flux-tunable transmon qubits where the qubit-qubit interaction is realized with a direct capacitive coupling. Utilizing this architecture it is possible to realize a flux-tuned |11>-|02> phase gate. An applied flux pulse that tunes the qubits in and out of the |11>−|02> avoided-crossing leads to a conditional phase accumulation. Leaving the system at the avoided-crossing for a specific time maps the state |11〉back into itself but acquires a minus sign in the process. As the computational states are far from being resonant with other transitions their phases evolve trivially and can be corrected using single qubit phase corrections and thus realize the CZ gate. The *baker.play* statement therefore contains a flux pulse that frequency-tunes transmon *q1* in and out of the avoided crossing |11>-|02> , while the *baker.frame_rotation_2pi* statements correct the single qubit phases.
+
+```python
+def bake_cz(baker: Baking, q0, q1):
+    q0_xy_element = f"qubit{q0}_xy"
+    q1_xy_element = f"qubit{q1}_xy"
+    q1_z_element = f"qubit{q1}_z"
+    baker.play("cz", q1_z_element)
+    baker.align()
+    baker.frame_rotation_2pi(qubit0_frame_update, q0_xy_element)
```

Comment:
What is qubit0_frame_update here? How do you calibrate it?
Why not putting it as an input parameter?

Reply:
@TheoLaudatQM please see this: https://github.com/qm-labs/qw-qm/pull/18
it is not a parameter of the CZ gate